### PR TITLE
docs: remote pinning services

### PR DESCRIPTION
**Pull request by jessicaschilling**
This PR makes room in docs for info related to pinning service integration in IPFS Desktop/Web UI:

- Adds new page at /how-to/work-with-pinning-services/
- Adds new page to left-hand nav
- Amends existing pinning page at /how-to/pin-files/

WIP! Using this PR to iterate on the content that will need to appear on these pages. Please do not merge until pinning service integration is ready to go live.

### CODE REVIEW

**Comment by jessicaschilling** [comment]
@lidel and @rafaelramalho19 -- here's a rough first pass at what I'm thinking the docs footprint of pinning service integration will be. Note that this is only one page, and that page addresses both how to add a pinning service in Desktop/Web UI _as well as_ roll your own pinning service -- which I suspect will be enough, since the guidance doesn't sound like it'll be much more detailed than "do what you want to, but be sure to obey the API spec". Would love to hear your thoughts on whether you agree/disagree at this point though, as well as any other thoughts in general. Preview [here](https://bafybeieh32ad76rzgzprdive7dsgcwn6mn42imw2q5pfo4j3jjgrhfwz54.ipfs.dweb.link/how-to/work-with-pinning-services/), plus amendments to existing pin page in docs [here](https://bafybeieh32ad76rzgzprdive7dsgcwn6mn42imw2q5pfo4j3jjgrhfwz54.ipfs.dweb.link/how-to/pin-files/).

@johnnymatthews -- just adding you for context at this point, so you know what's in the pipeline. 😊

**Comment by jessicaschilling** [comment]
@lidel - sounds good!

@rafaelramalho19 - I think we're therefore safe to use https://docs.ipfs.io/how-to/work-with-pinning-services/ as our docs link 😊

File: docs/how-to/pin-files.md
Comment (johnnymatthews) [inline docs/how-to/pin-files.md:48]:
```suggestion
## Local versus remote pinning
```

Trying to avoid abbreviations as much as possible, they might cause issues with translation efforts down the road.

**Review by johnnymatthews — DISMISSED** [review]

**Comment by johnnymatthews** [comment]
Just one tiny change and then it's all good from me!

**Comment by johnnymatthews** [comment]
@jessicaschilling is this ready to be merged in?

**Comment by jessicaschilling** [comment]
Not for a bit yet 😊 will keep you posted!

**Comment by johnnymatthews** [comment]
No worries. I'm just gonna mark this PR as a draft for now then :)

**Comment by jessicaschilling** [comment]
@johnnymatthews @lidel @autonome -- this isn't quite ready to ship but requesting your review on the text so we can keep this moving in preparation for go-live.

IMPORTANT NOTE: Images are just FPO from this autumn's mockups, and will be replaced with real screenshots once we've finished wiring up ipfs-webui.

File: docs/how-to/pin-files.md
Comment (johnnymatthews) [inline docs/how-to/pin-files.md:50]:
```suggestion
All the information above assumes that you're pinning items locally — that is, to your local IPFS node. That's the default behavior for IPFS, but it's also possible to pin your files to a _remote pinning service_. These third-party services give you the opportunity to pin files not to your own local node, but to nodes that they operate. You don't need to worry about your own node's available disk space or uptime.
```

File: docs/how-to/pin-files.md
Comment (johnnymatthews) [inline docs/how-to/pin-files.md:52]:
```suggestion
While you can use a remote pinning service's own GUI, CLI, or other dev tools to manage IPFS files pinned to their service, you can also work directly with pinning services using your local IPFS installation — meaning that you don't need to learn a pinning service's unique API or other tooling.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:8]:
```suggestion
Depending on how you use IPFS, you might find it helpful to use a **remote pinning service** instead of, or in addition to, pinning files on your local IPFS node. Whether it happens remotely or locally, **pinning** an item in IPFS identifies it as something you always wish to keep available, exempting it from the routine _garbage collection_ that IPFS does on infrequently-used items in order to efficiently manage storage space. [Learn more about pinning →](/how-to/pin-files).
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:10]:
```suggestion
If you've got just one local IPFS node that's always running, local pinning may be all you need to ensure your important items are persisted and never garbage-collected. However, using a remote pinning service — or creating your own — might be useful to you if:
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:15]:
```suggestion
- Your local node isn't always online, but you need items to be consistently available
- You'd like to keep a persistent backup of your local node's files somewhere else.
- You don't have all the disk space you need on your local node.
- You run more than one IPFS node, and would like to use one of them as a "personal pinning service" as your preferred location for permanent storage.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:17]:
```suggestion
There are a number of commercial pinning services that make it easy for you to purchase pinning capacity for your important files, some of which include Pinata, Temporal, Infura, and others. Each of these third-party services has its own unique interface for pinning files and managing those pins; this could include a GUI, an API, CLI commands, or other tooling.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:19]:
```suggestion
However, you don't need to learn new commands or tools if your pinning service of choice supports the [IPFS Pinning Service API](https://ipfs.github.io/pinning-services-api-spec/) specification — because those pinning services are supported within IPFS itself through the command line, [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop), and the [IPFS Web UI](https://github.com/ipfs-shipyard/ipfs-webui). And, if you're interested in creating your own pinning service for your own personal or shared use, you can directly integrate it with IPFS Desktop/Web UI and the IPFS CLI by using the IPFS Pinning Service API.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:25]:
```suggestion
To add and use a remote pinning service directly in IPFS, you'll first need to have an account with that service. Once you've got an account, follow these steps to add and use it:
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:33]:
```suggestion
To add a new pinning service, open up IPFS Desktop or Web UI, navigate to the **Pinning Services** section of the **Settings** screen, and click the **Add Service** button:
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:37]:
```suggestion
Then, select your chosen pinning service from the modal that pops up. If the pinning service you'd like to add isn't listed in that modal, don't worry — you can add any remote pinning service that supports the IPFS Pinning Service API by clicking the **Add a custom one** link.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:43]:
```suggestion
- A nickname for this service. Tthis can be helpful if, for example, you want to add two accounts from the same service.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:44]:
```suggestion
- The URL for your service's API endpoint. This field only appears if you've selected a custom pinning service!
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:45]:
```suggestion
- Your secret API key. This is the unique token provided to you by the pinning service — check its documentation for more info.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:46]:
```suggestion
- Whether you want to automatically add all files in your local node’s Files directory to the pinning service, or choose which ones you upload.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:50]:
```suggestion
After you hit **Save**, you’ll see your new pinning service added to the **Pinning Services** section of your **Settings** screen.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:56]:
```suggestion
Now that you’re set up, you can pin or unpin files to your new pinning service directly from the Files screen. Just right-click any file or click the **three dots** action icon in the files list, and select **Set pinning**:
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:74]:
```suggestion
- `nickname` is a unique name for this particular instantiation of a pinning service. TThis can be helpful if, for example, you want to add two accounts from the same service.
- `https://api.mypinningservice.com/endpoint` is the endpoint supplied to you by the pinning service. Check the service's documentation for more info.
- `myAccessToken` is the unique token provided to you by the pinning service. Check the service's documentation for more info.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:108]:
```suggestion
- Designating one of your own IPFS nodes to be a _personal pinning service_ as a preferred location for permanent storage.
```

File: docs/how-to/work-with-pinning-services.md
Comment (johnnymatthews) [inline docs/how-to/work-with-pinning-services.md:110]:
```suggestion
- Running a private pinning service for your friends or company.
- Starting your own commercial pinning service.
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]

**Comment by johnnymatthews** [comment]
Made a few changes, mainly formatting. 

Stuff the reader needs to click on or read should be in bold: 

- Click **Pin file**.
- Go to **Settings** and click **Reset**.

I tend to stay away from using parenthesis as much as possible. They don't play well with language translators, and can often be split up into multiple sentences pretty easily. Also, _via_ is a weird word that language translators can't deal with, so I tend to use _through_ or _using_, stuff like that.

Outside of that, I like this PR! Merge when ready.

**Comment by jessicaschilling** [comment]
Thanks for the detailed review, @johnnymatthews! Will remember the overall guidance for future work - thanks for the info.

**Comment by lidel** [comment]
@jessicaschilling I made a pass at this in https://github.com/ipfs/ipfs-docs/pull/369/commits/cb629c73dcc9dc33e042957a484dff828f87d26a 
- hidden GUI sections  –  blocked by https://github.com/ipfs/ipfs-gui/issues/91
- updated CLI a bit to reflect go-ipfs 0.8.0-rc2
- linked to guidance how to generate own service from OpenAPI spec

**Review by lidel — APPROVED** [review]
LGTM, but we need to confirm link to Pinata docs works before merging this.
cc @obo20

**Comment by jessicaschilling** [comment]
@lidel In hindsight this was probably not the best way for me to do it, but I left a few comments on your commit https://github.com/ipfs/ipfs-docs/commit/cb629c73dcc9dc33e042957a484dff828f87d26a. Let me know if you'd rather me just duplicate those as a review.

One note: @johnnymatthews has a periodic release cadence going on for docs now, so ping him before you merge. 😊  Thank you!

**Comment by lidel** [comment]
@jessicaschilling comments on commit were fine, thank you! 

I removed mention of "porcelain" – indeed this is a bit niche technical term (as in [Git Internals - Plumbing and Porcelain](https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain)), better to not add to the pile :)

Mind eyeballing my changes in https://github.com/ipfs/ipfs-docs/pull/369/commits/5896c56a6cea9dc027650b5361ef6e1e68e41b91 before I summon Johnny?

**Comment by jessicaschilling** [comment]
@lidel LGTM, thank you!

**Review by lidel — APPROVED** [review]

**Review by johnnymatthews — APPROVED** [review]

**Comment by lidel** [comment]
@johnnymatthews https://github.com/ipfs/go-ipfs/releases/v0.8.0 is tagged and people will start looking into this now.

Mind merging?

**Pull request merged**
